src/App.jsx

```diff
@@ -14,33 +17,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const isArraySame = (arr1, arr2) => {
+  if (arr1.length !== arr2.length) {
+    return false;
+  }
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  return arr1.every((element, index) => element === arr2[index]);
+};
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+export const clear = (...args) => {
+  args.forEach(arg => arg(false));
+};
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortOrder, setSortOrder] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortOrder) {
+      case 'alphabetical':
+        return good1.localeCompare(good2);
+      case 'length':
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortOrder !== 'alphabetical',
+          })}
+          onClick={() => setSortOrder('alphabetical')}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortOrder !== 'length',
+          })}
+          onClick={() => setSortOrder('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
+          Reverse
+        </button>
+
+        {!isArraySame(goodsFromServer, sortedGoods) && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={() => clear(setSortOrder, setReversed)}
```

Comment:
looks strange, you don't want to set false in setSortOrder state, because it's accept only string values

Reply:
better to write it as usual